src/lib.rs

```diff
@@ -199,3 +210,65 @@ fn create_http_client(
 
     relay_client::http::Client::new(&conn_opts).unwrap()
 }
+
+fn new_geoblocking_router<S, B>(
+    geoip_resolver: Option<Arc<MaxMindResolver>>,
+    blocked_countries: Vec<String>,
+) -> Router<S, B>
+where
+    S: Clone + Send + Sync + 'static,
+    B: HttpBody + Send + 'static,
+{
+    if let Some(resolver) = geoip_resolver {
+        Router::new().layer(GeoBlockLayer::new(
+            resolver.clone(),
+            blocked_countries.clone(),
+            BlockingPolicy::AllowAll,
+        ))
+    } else {
+        Router::new()
+    }
+}
+
+async fn get_s3_client(config: &Configuration) -> S3Client {
+    let region_provider = RegionProviderChain::first_try(Region::new("eu-central-1"));
+    let shared_config = aws_config::from_env().region(region_provider).load().await;
+
+    let aws_config = match &config.s3_endpoint {
+        Some(s3_endpoint) => {
+            info!(%s3_endpoint, "initializing analytics with custom s3 endpoint");
+
+            aws_sdk_s3::config::Builder::from(&shared_config)
+                .endpoint_url(s3_endpoint)
+                .build()
+        }
+        _ => aws_sdk_s3::config::Builder::from(&shared_config).build(),
+    };
```

Comment:
```suggestion
    let mut builder = aws_sdk_s3::config::Builder::from(&shared_config);
    if let Some(s3_endpoint) = &config.s3_endpoint {
        info!(%s3_endpoint, "initializing analytics with custom s3 endpoint");
        builder = builder.endpoint_url(s3_endpoint);
    };
    aws_config = builder.build();
```

Reply:
Same result, I do like the match

Reply:
Seems like everyone has an opinion on this across services :smile: 
Another reason to have it in `utils-rs`

I do like the @chris13524 's version more though

Ideally, the builder should have an API that supports both `T` and `Option<T>` for each configuration option. But for some reason no one wants to _build their builders_ in such a way

Reply:
It's a bit like traits and Arcs/Refs, easy to forget and kinda stupid that there's no easy  way to say "that works for refs too, trust me"

Reply:
As Ivan said, let's stop bikeshedding and roll with that, then push it to `utils-rs` later.